src/relations/postgresql_provider.py

```diff
@@ -265,8 +265,9 @@ def update_tls_flag(self, tls: str) -> None:
             ca = ""
 
         for relation in relations:
-            self.database_provides.set_tls(relation.id, tls)
-            self.database_provides.set_tls_ca(relation.id, ca)
+            if self.database_provides.fetch_relation_field(relation.id, "database"):
+                self.database_provides.set_tls(relation.id, tls)
+                self.database_provides.set_tls_ca(relation.id, ca)
```

Comment:
Check if the relation is already properly intialised. If not the values should be set by the `database_requested` hook.